lsp/server/src/diagnostic/DiagnosticProducer.ts

```diff
@@ -19,4 +26,9 @@ export interface DiagnosticProducer<T> {
         textDocument: TextDocument,
         astNode: T
     ): Promise<Diagnostic[]>;
+
+}
+
+export interface DiagnosticMetaData {
```

Comment:
Diagnostic sent from server to client could have data?: Any.  we utilize it to send the producer id, so CodeAction could implement a quickfix to suppress that producer.